tree_segment.py

```diff
@@ -0,0 +1,434 @@
+import laspy
+import matplotlib.pyplot as plt
+import math
+import numpy as np
+import random
+import time
+import pandas as pd
+import argparse
+import re
+import os
+
+
+class Point_Cloud:
+    # object that stores the points in a point cloud, initialized from a las file
+    # points can be enabled or disabled, disabled points are ignored when searching for stems
+    def __init__(self, file_name, verbose=False):
+        all_data = laspy.read(file_name)
+        height_of_points_above_ground = np.array(
+            [point[-1] for point in all_data.points.array])
+        self.points = [Point(xyz, height) for xyz, height in zip(
+            all_data.xyz, height_of_points_above_ground)]
+        # internal, to access this use enabled_points()
+        self.__enabled_points__ = self.points
+        self.verbose = verbose
+
+    def enabled_points(self):
+        # filters the set of points for just those enabled, then returns that set of enabled points
+        self.__enabled_points__ = filter_for_enabled(self.__enabled_points__)
```

Comment:
Nit: Afaik, this is always recomputed before usage. Consider removing the internal representation and returning the result of the computation directly.

Reply:
I think this is a feature, not a bug? My idea was that iff there are 10k points, but only 3k enabled, we save time by remembering that the 7k have already been disabled and we don't have to check them again when we run this method.

I do think running the filter every time we want the list of enabled points might be wasteful. Maybe I should only be updating the enabled list when I disable points? I think that's only in disable_stem_region(), so I can just recompute _enabled_points at the end of that?

Reply:
Okay, yes I did this and there isn't a significant speedup but it does remove an unnecessary piece of complexity!